test/SFSmartStoreTestSuite.js

```diff
@@ -959,7 +959,7 @@ SmartStoreTestSuite.prototype.testFullTextSearchAgainstArrayNode  = function() {
             return self.upsertSoupEntries(soupName,rawEntries);
         })
         .pipe(function(entries) {
-            var querySpec = navigator.smartstore.buildMatchQuerySpec("attributes.color", "grey");
+            var querySpec = navigator.smartstore.buildMatchQuerySpec("attributes.color", "grey", "descending", 10, "name");
```

Comment:
If we don't specify the order path, it will sort by the query path - in this case a non-leaf node.